test/integration/webhook/workload_test.go

```diff
@@ -117,6 +150,325 @@ var _ = ginkgo.Describe("Workload validating webhook", func() {
 			ginkgo.Entry("invalid podSet.Count", 3, 0, true),
 			ginkgo.Entry("valid podSet", 3, 3, false),
 		)
+
+		ginkgo.DescribeTable("Should have valid values when creating", func(w func() *kueue.Workload, errorType int) {
+			err := k8sClient.Create(ctx, w())
+			switch errorType {
+			case isForbidden:
+				gomega.Expect(err).Should(gomega.HaveOccurred())
+				gomega.Expect(err).Should(testing.BeAPIError(testing.ForbiddenError), "error: %v", err)
+			case isInvalid:
+				gomega.Expect(err).Should(gomega.HaveOccurred())
+				gomega.Expect(err).Should(testing.BeAPIError(testing.InvalidError), "error: %v", err)
+			default:
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			}
+		},
+			ginkgo.Entry("workload",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).PodSets(
+						*testing.MakePodSet("driver", 1).Obj(),
+						*testing.MakePodSet("workers", 100).Obj(),
+					).Obj()
+				},
+				isValid),
+			ginkgo.Entry("podSet name",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).PodSets(
+						*testing.MakePodSet("@driver", 1).Obj(),
+					).Obj()
+				},
+				isInvalid),
+			ginkgo.Entry("priorityClassName",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).
+						PriorityClass("invalid_class").
+						Priority(0).
+						Obj()
+				},
+				isInvalid),
+			ginkgo.Entry("priorityClassName is empty",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).
+						Obj()
+				},
+				isValid),
+			ginkgo.Entry("priorityClassName is set",
+				func() *kueue.Workload {
+					return testing.MakeWorkload(workloadName, ns.Name).
+						PriorityClass("priority").
+						Obj()
+				},
+				isInvalid),
+			ginkgo.Entry("queueName",
```

Comment:
```suggestion
			ginkgo.Entry("invalid queueName",
```

same for other tests if it's the case